File: Sangeun/src/week1/week8_problem_2.py
# https://programmers.co.kr/learn/courses/30/lessons/12943
def collatz(cur_num, seq):
    if seq >= 500:
        return -1
    if cur_num == 1:
        return seq
    if cur_num % 2 == 0:
        cur_num = cur_num // 2
    else:
        cur_num = cur_num * 3 + 1

    return collatz(cur_num, seq + 1)

def solution(num):
    answer = collatz(num, 0)
    return answer

File: Shinhang/src/week1/week8_problem_2.java
package week1;

class Solution {
    public int solution(int num) {
        int answer = 0;
        long number = num;
        while(true){
            if(number == 1 || answer == 500){
                break;
            }
            if((number & 1) == 1){
                number *= 3;
                number++;
            } else {
                number /= 2;
            }
            answer++;
        }
        return (answer != 500 ? answer : -1);
    }
}